### DISCUSSION TITLE
On @babel/eslint-parser, what does allowImportExportEverywhere mean exactly?

### DISCUSSION THREAD

**Discussion by branlok** [babel/babel#13864] [Q&A] [answered]
the document says:
"allowImportExportEverywhere (default false) can be set to true to allow import and export declarations to appear anywhere a statement is allowed if your build environment supports that. Otherwise import and export declarations can only appear at a program's top level."

I've set this false, in my javascript file i have `import peko from "./assets/peko.jpg";` I tried playing around and move this line from top to the bottom of my js file, then running build it didnt not show any error.

can someone clarify what this option does?

in my .eslintrc.js it contains 
```
module.exports = {
  parser: "@babel/eslint-parser",
  parserOptions: {
    sourceType: "module",
    allowImportExportEverywhere: false,
    ecmaFeatures: {
      globalReturn: false,
    },
    babelOptions: {
      configFile: "./webpack.config.js",
    },
  },
};
```

**Comment by nicolo-ribaudo** [comment, marked as answer]
When it's set to true, it will allow using (invalid) code like this:
```js
if (test) {
  import "dependency-test";
} else {
  import "dependency-prod";
}
```

This can be useful if, for example, you have a Babel plugin that replaces that `if` statement with a specific branch depending on the environment.
